src/datasource.ts

```diff
@@ -39,26 +38,45 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
           { name: 'id', type: FieldType.string },
         ],
       });
-      await response.then((response) => {
-        response.data.incidents.forEach((incident: Incident) => {
-          const timestamp: Date = new Date(incident.created_at);
-          const timestamp_end: Date = new Date(incident.resolved_at);
-          const createdAt = timestamp.getTime();
-          const resolvedAt = timestamp_end.getTime();
-          const title = incident.title;
-          const text = incident.summary;
-          const id = incident.incident_key;
-
-          frame.appendRow([createdAt, resolvedAt, title, text, id]);
-        });
+      const response = await this.doRequestPaginated(target, from, to, 25);
```

Comment:
It doesn't make much sense to call this function with `from` and `to`. They will always be the same. Are you envisioning a case where they won't in the nearby future?

Reply:
`from` and `to` are respectively the edges of the time frame queried, so they'll change when the time frame is changed in Grafana, and used later on when calling the PagerDuty API to get incidents only for the specified time frame in the `doRequest` function. I'm not sure I get what you mean by 

> They will always be the same. 

You mean in the sense they're going to be the same at each iteration of the pagination routine?

Reply:
Ignore this, please.